### PR TITLE
Add ksm merge-across-nodes

### PR DESCRIPTION
**Pull request by chrisho**
Problem:
User machines need to support KSM merge-across-nodes

Solution:
Add KSM merge-across-nodes

Related Issue:
https://github.com/harvester/harvester/issues/2827

Test plan:
1. Toggle ksmtuned merge_across_nodes, `kubectl edit ksmtuned node1`, update **spec.ksmtunedParameters.MergeAcrossNodes** to 0 or 1.
2. Wait for a few minutes, `cat /sys/kernel/mm/ksm/merge_across_nodes` should display you want.

### CODE REVIEW

File: pkg/controller/ksmtuned/controller.go [hunk 1]
```diff
@@ -82,6 +82,11 @@ func (c *Controller) OnChange(key string, kt *ksmtunedv1.Ksmtuned) (*ksmtunedv1.
 		return kt, nil
 	}
 
+	// process merge across nodes changed
+	if err := c.Ksmtuned.ToggleMergeAcrossNodes(kt.Spec.KsmtunedParameters.MergeAcrossNodes); err != nil {
+		return kt, err
+	}
+
```
Comment (guangbochen) [inline pkg/controller/ksmtuned/controller.go:89]: should we only config this when the ksmd is running?

File: pkg/apis/node.harvesterhci.io/v1beta1/ksmtuned.go [hunk 2]
```diff
@@ -107,4 +107,7 @@ type KsmtunedParameters struct {
 	Decay     uint   `json:"decay"`
 	MinPages  uint   `json:"minPages"`
 	MaxPages  uint   `json:"maxPages"`
+
+	// +kubebuilder:validation:Maximum=1
+	MergeAcrossNodes uint `json:"mergeAcrossNodes"`
```
Comment (guangbochen) [inline pkg/apis/node.harvesterhci.io/v1beta1/ksmtuned.go:112]: users won't be able to modify it if the parameters are not displayed? should we move this to the upper spec?

File: pkg/ksmtuned/ksmtuned.go [hunk 3]
```diff
@@ -258,6 +260,31 @@ func (k *Ksmtuned) RunStatus() (ksmtunedv1.KsmdPhase, error) {
...
+// ToggleMergeAcrossNodes toggle /sys/kernel/mm/ksm/merge_across_nodes
+func (k Ksmtuned) ToggleMergeAcrossNodes(toggle uint) error {
+	mergeAcrossNodesLocker.Lock()
+	defer mergeAcrossNodesLocker.Unlock()
+	if unchanged, err := k.compareMergeAcrossNodes(toggle); err != nil {
+		return err
+	} else if unchanged {
+		return nil
+	}
+	return k.ksmd.toggleMergeAcrossNodes(toggle)
```
Comment (guangbochen) [inline pkg/ksmtuned/ksmtuned.go:285]: should pass the context to the `k.ksmd.toggleMergeAcrossNodes`

File: pkg/controller/ksmtuned/controller.go (hunk 1 above)
Comment (chrisho) [reply pkg/controller/ksmtuned/controller.go:89]: when ksmd running, pages are shared, so unable to config merge across nodes.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]
LGTM, thanks for adding this feature.

**Review by futuretea — APPROVED** [review]
LGTM

**Pull request merged**